File: stack/9012/jinho.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const datas = input.slice(1).map(e => e.trim());

for(let i=0; i<datas.length; i++){
    let data = datas[i];
    let isVPS = true;
    while(data.length){
        if(data.length===1){
            isVPS = false;
            break;
        }
        const idx = data.indexOf('()');
        if(data.length>=2 && idx===-1){
            isVPS = false;
            break;
        }
        data = data.split('');
        data.splice(idx, 2);
        data = data.join('')
    }
    if(isVPS) console.log('YES');
    else console.log('NO');
}

File: greedy/1541/jinho.py
import sys

# data = list(map(int, sys.stdin.readline().split()))
input = sys.stdin.readline().rstrip()
data = input.split("-")
_data = list(map(lambda x : list(map(int, x.split("+"))),data))

sum = 0
for _ in range(len(_data[0])):
    sum += _data[0][_]

for i in range(1, len(_data)):
    for j in range(len(_data[i])):
        sum -= _data[i][j]

print(sum)